### PR TITLE
Bringing scan-build in RTR tool changes back from 5.0 into master

### LINKED ISSUE #9917: Bring RTR scanbuild execution changes from 5.0 branch
## Description
This issue aims to bring changes being done as part of the issue [#8045](https://github.com/wazuh/wazuh/issues/8045) into the current branch to get scanbuild executed automatically as part of the RTR CI checks.
**Related PR:** https://github.com/wazuh/wazuh/pull/8289

## DoD:
- [x] Adapt PR changes
- [x] CI checks
- [x] Create PR

### PR DESCRIPTION
**Pull request by tinachoB**
|Related issue|
|---|
|Closes #9917 |

## Description

This PR aims to bring changes being done as part of the issue [#8045](https://github.com/wazuh/wazuh/issues/8045) into the current branch to get scanbuild executed automatically as part of the RTR CI checks.
**Related PR:** https://github.com/wazuh/wazuh/pull/8289## Configuration options

This will run for TARGET: `agent`, `server `and `winagent`
image

## Tests

- Compilation without warnings in every supported platform
  - [X] Linux
  - [X] Windows
  - [ ] MAC OS X
- [ ] Source installation
- [ ] Package installation
- [ ] Source upgrade
- [ ] Package upgrade
- [ ] Review logs syntax and correct language
- [ ] QA templates contemplate the added capabilities

- Memory tests for Linux
  - [X] Scan-build report
  - [ ] Coverity
  - [ ] Valgrind (memcheck and descriptor leaks check)
  - [ ] Dr. Memory
  - [ ] AddressSanitizer
- Memory tests for Windows
  - [X] Scan-build report
  - [ ] Coverity
  - [ ] Dr. Memory
- Memory tests for macOS
  - [ ] Scan-build report
  - [ ] Leaks
  - [ ] AddressSanitizer

- [ ] Retrocompatibility with older Wazuh versions
- [ ] Working on cluster environments
- [ ] Configuration on demand reports new parameters
- [ ] The data flow works as expected (agent-manager-api-app)
- [ ] Added unit tests (for new features)
- [ ] Stress test for affected components

- Decoder/Rule tests
  - [ ] Added unit testing files ".ini"
  - [ ] runtests.py executed without errors

### CODE REVIEW

File: src/ci/utils.py [hunk 1]
```diff
@@ -446,6 +510,42 @@ def runASAN(moduleName):
...
+        if out.returncode == 1:
+            print(out.stdout)
+        else:
+            print(out.stderr)
+        printFail('[SCANBUILD: FAILED]')
+        errorString = 'Error Running Scan-build: ' + str(out.returncode)
+        raise ValueError(errorString)
+
+    printGreen("<"+targetName+">"+"[SCANBUILD: PASSED]"+"<"+targetName+">")
+
```
Comment (jnasselle) [inline src/ci/utils.py:548]:
```suggestion


```

File: src/ci/utils.py [hunk 2]
```diff
@@ -364,6 +365,69 @@ def cleanFolder(moduleName, additionalFolder):
...
+    out = subprocess.run(makeTargetCommand, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
+    if out.returncode == 0:
+        printGreen('[MakeTarget: PASSED]')
+    else:
+        print(makeTargetCommand)
+        print(out.stderr)
+        printFail('[MakeTarget: FAILED]')
+        errorString = 'Error Running MakeTarget: ' + str(out.returncode)
+        raise ValueError(errorString)
```
Comment (jnasselle) [inline src/ci/utils.py:430]:
```suggestion
        raise ValueError(errorString)

```

File: src/ci/utils.py [hunk 3]
```diff
@@ -364,6 +365,69 @@ def cleanFolder(moduleName, additionalFolder):
...
+        errorString = 'Error Running MakeDeps: ' + str(out.returncode)
+        raise ValueError(errorString)
+
+def makeTarget(targetName, tests, debug):
+    makeTargetCommand = "make TARGET=" + targetName
+    if tests:
+        makeTargetCommand += " TEST=1"
+    if debug:
+        makeTargetCommand += " DEBUG=1"
+
```
Comment (jnasselle) [inline src/ci/utils.py:421]:
```suggestion


```

File: src/ci/utils.py [hunk 4]
```diff
@@ -364,6 +365,69 @@ def cleanFolder(moduleName, additionalFolder):
...
+    out = subprocess.run(makeDepsCommand, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
+    if out.returncode == 0:
+        printGreen('[MakeDeps: PASSED]')
+    else:
+        print(makeDepsCommand)
+        print(out.stderr)
+        printFail('[MakeDeps: FAILED]')
+        errorString = 'Error Running MakeDeps: ' + str(out.returncode)
+        raise ValueError(errorString)
+
```
Comment (jnasselle) [inline src/ci/utils.py:414]:
```suggestion


```

File: src/ci/utils.py [hunk 5]
```diff
@@ -364,6 +365,69 @@ def cleanFolder(moduleName, additionalFolder):
...
+    out = subprocess.run("rm -rf ./external/*", stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
+    if out.returncode == 0 and not out.stderr:
+        printGreen('[CleanExternals: PASSED]')
+    else:
+        print("rm -rf ./external/*")
+        print(out.stderr)
+        printFail('[CleanExternals: FAILED]')
+        errorString = 'Error Running CleanExternals: ' + str(out.returncode)
+        raise ValueError(errorString)
+
```
Comment (jnasselle) [inline src/ci/utils.py:400]:
```suggestion


```

File: src/ci/utils.py [hunk 6]
```diff
@@ -364,6 +365,69 @@ def cleanFolder(moduleName, additionalFolder):
...
+    out = subprocess.run("make clean-internals", stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
+    if out.returncode == 0:
+        printGreen('[CleanInternals: PASSED]')
+    else:
+        print("make clean-internals")
+        print(out.stderr)
+        printFail('[CleanInternals: FAILED]')
+        errorString = 'Error Running CleanInternals: ' + str(out.returncode)
+        raise ValueError(errorString)
+
```
Comment (jnasselle) [inline src/ci/utils.py:389]:
```suggestion


```

File: src/ci/utils.py [hunk 7]
```diff
@@ -364,6 +365,69 @@ def cleanFolder(moduleName, additionalFolder):
...
+    out = subprocess.run("make clean", stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
+    if out.returncode == 0:
+        printGreen('[CleanAll: PASSED]')
+    else:
+        print("make clean")
+        print(out.stderr)
+        printFail('[CleanAll: FAILED]')
+        errorString = 'Error Running CleanAll: ' + str(out.returncode)
+        raise ValueError(errorString)
+
```
Comment (jnasselle) [inline src/ci/utils.py:378]:
```suggestion


```

**Review by jnasselle — COMMENTED** [review]
Hi @tinachoB !

Nice work here! I only added some suggestions related to PEP8 need for two blank spaces between functions.

Regards,
Nico

**Review by HanesSciarrone — APPROVED** [review]

**Review by jnasselle — DISMISSED** [review]

**Review by HanesSciarrone — APPROVED** [review]

**Review by jnasselle — APPROVED** [review]

**Review by Dwordcito — APPROVED** [review]
Good Job!

**Comment by tinachoB** [comment]
@jnasselle , @HanesSciarrone , @Dwordcito and @vikman90 - These changes are working as expected using scan-build-10 version. The Docker used to execute scan build for target: agent, winagent and server is updated with this scan-build version as well: Reference [here](https://github.com/wazuh/wazuh-jenkins/blob/master/dockers/util/test_RTR/Dockerfile#L20).

**Final results:**
image

From now on, with these changes, every single branch being created from master will have these changes. This will help to have master branch consistent with scan-build fixes.

**Pull request merged**